### PR TITLE
ci: Add testing on Python 3.11 [WIP]

### PR DESCRIPTION
**Pull request by rht**

### CODE REVIEW

**Comment by rht** [comment]
Test failed on 3.11. I will fix this later.

File: .github/workflows/build_lint.yml [hunk 1]
```diff
@@ -32,6 +32,8 @@ jobs:
         os: [windows, ubuntu, macos]
         python-version: ["3.10"]
         include:
+          - os: ubuntu
+            python-version: "3.11"
```
Comment (EwoutH) [inline .github/workflows/build_lint.yml:36]: Thanks, that needed to be done indeed! I would say let's bump the version that runs on all OSes to 3.11, and add a 3.10 job for Ubuntu.
```suggestion
        python-version: ["3.11"]
        include:
          - os: ubuntu
            python-version: "3.10"
```
Do you have any idea what the error is about? It doesn't look that Python 3.11 specific.

Comment (rht) [reply .github/workflows/build_lint.yml:36]: >  I would say let's bump the version that runs on all OSes to 3.11, and add a 3.10 job for Ubuntu.

Will do, though I'm still investigating. I have reproduced the bug in a Docker instance.

Comment (Tortar) [reply .github/workflows/build_lint.yml:36]: I found the source of the bug, but I'm not sure why the other Python didn't identify it as a bug actually

Comment (Tortar) [reply .github/workflows/build_lint.yml:36]: If you want to add the mod to this PR, I will close mine

Comment (rht) [reply .github/workflows/build_lint.yml:36]: Let's solve it in #1521, and limit the scope of this PR to updating the CI.

**Comment by jackiekazil** [comment]
@rht are we waiting for resolution on #1521 & merge both at once? (I could go either way... I am leaning towards merging this, but wanted to check in and see if you were in the same head space.)

**Comment by rht** [comment]
I need to understand why the problem happens in 3.11, on whether it is specific to `unittest.mock.patch`, or it could happen again in other context, due to the conflicting import namespace.

**Comment by Tortar** [comment]
Maybe trying it again with 3.11.1 bugfix release?

**Comment by rht** [comment]
It works on 3.11.1! :tada:

**Comment by EwoutH** [comment]
Awesome! One of the first times I have seen a bug actually from CPython itself.

File: .github/workflows/build_lint.yml (hunk 1 above)
Comment (EwoutH) [reply .github/workflows/build_lint.yml:36]: > Thanks, that needed to be done indeed! I would say let's bump the version that runs on all OSes to 3.11, and add a 3.10 job for Ubuntu.

@rht What do you think of this?

Comment (rht) [reply .github/workflows/build_lint.yml:36]: Sounds good. Doing it now.

**Review by EwoutH — APPROVED** [review]
Awesome, LGTM!

**Pull request merged**